### PR TITLE
storm - tree() operator

### PR DESCRIPTION
**Pull request by vEpiphyte**
Add a tree() operator for storm for doing pivots on type recursive forms.

### CODE REVIEW

File: docs/synapse/userguides/ug015_storm_ref_pivot.rst [hunk 1]
```diff
@@ -33,6 +33,52 @@ jointags()
...
+
+  # Destination only form - traversing all of the woot.com children nodes.
+  inet:fqdn = woot.com tree( inet:fqdn:domain )
+
+  # Select the entire syn:tage=foo tree.
+  syn:tag=foo tree(syn:tag, syn:tag:up)
+
+  # tree() up - select all parent fqdns of mx.somebox.woot.com
+  inet:fqdn = mx.somebox.woot.com tree( inet:fqdn:domain, inet:fqdn )
+
```
Comment (therealsilence) [inline docs/synapse/userguides/ug015_storm_ref_pivot.rst]: For consistency with other docs, consider adding a bold header for:

**Usage Notes**

After examples and before your "notes".
Similarly, consider making individual notes bullet points.

Really just a formatting nitpick, it's good either way!

Comment (vEpiphyte) [reply docs/synapse/userguides/ug015_storm_ref_pivot.rst]: fixed these up! thanks.

File: synapse/lib/storm.py [hunk 2]
```diff
@@ -1414,3 +1415,65 @@ def _stormOperTask(self, query, oper):
...
+    def _stormOperTree(self, query, oper):
+
+        args = oper[1].get('args')
+        opts = dict(oper[1].get('kwlist'))
+
+        # Prevent bad recursions????
+        recurlim = opts.get('recurlim', 20)
+
+        srcp = None
+        dstp = args[0]
```
Comment (vertexmc) [inline synapse/lib/storm.py]: What if there are no args?

File: synapse/lib/storm.py [hunk 3]
```diff
@@ -1414,3 +1415,65 @@ def _stormOperTask(self, query, oper):
...
+            if not qvals:
+                break
+
+            [query.add(t) for t in self.stormTufosBy('in', dstp, qvals, limit=opts.get('limit'))]
+
+            queried_vals = queried_vals.union(vals)
+
+            if recurlim:
+                recurlim -= 1
+                if not recurlim:
```
Comment (vertexmc) [inline synapse/lib/storm.py]: Might make more sense to replace this with `if recurlim < 1:`

File: synapse/lib/storm.py (hunk 2 above)
Comment (vEpiphyte) [reply synapse/lib/storm.py]: Added a bad syntax error in the event of empty args.

File: synapse/lib/storm.py (hunk 3 above)
Comment (vEpiphyte) [reply synapse/lib/storm.py]: done; and added a type norm step to ensure we're working with a "int" value.

**Pull request merged**